Mobile/ShopLens/Droid/Helpers/ShopLensSpeechRecognizer.cs

```diff
@@ -0,0 +1,40 @@
+﻿using System;
+using Android.App;
+using Plugin.SpeechRecognition;
+
+namespace ShopLens.Droid.Helpers
+{
+    public class ShopLensSpeechRecognizer
+    {
+        public event EventHandler<ShopLensSpeechRecognizedEventArgs> OnPhraseRecognized;
+
+        public ShopLensSpeechRecognizer(Action<object, ShopLensSpeechRecognizedEventArgs> ReactToSpeechInput)
+        {
+            OnPhraseRecognized += (obj, eargs) => ReactToSpeechInput(obj, eargs);
+        }
+
+        public void RecognizePhrase(Activity activity)
+        {
+            // According to Java, Voice Recognizer needs to be run on the Main Ui Thread.
+            activity.RunOnUiThread(ListenForAPhrase);
```

Comment:
Got a Java.RuntimeException from this method, and the expection pop up told me to have voice recognizer run on the Main thread, so that's what I did here.

Reply:
Well we really don't have a choice in the matter...

Reply:
It doesn't block the UI thread, I tested it today 😃.